### ISSUE TITLE
Infinite loop when entering too long ping filter

### ISSUE THREAD

**Issue by ChillerDragon** [ddnet/ddnet#3322]
In server browser when setting Maximum ping in filter to a string longer than br_filter_ping the client hangs in a infinite loop.

https://github.com/ddnet/ddnet/blob/a9631bf9cd71bb4e7b6a67e71976a7c9eace8bd5/src/engine/client/text.cpp#L941

```
Thread 1 "DDNet" received signal SIGINT, Interrupt.
0x00005555556d538c in CTextRender::TextEx (this=0x7ffff7fa9010, pCursor=0x7fffffffa960, pText=0x7fffffffae73 "999", Length=4)
    at /home/chiller/Desktop/git/ddnet/src/engine/client/text.cpp:947
947                                     int Wlen = minimum(WordLength(pCurrent), (int)(pEnd - pCurrent));
(gdb) bt
#0  0x00005555556d538c in CTextRender::TextEx(CTextCursor*, char const*, int)
    (this=0x7ffff7fa9010, pCursor=0x7fffffffa960, pText=0x7fffffffae73 "999", Length=4)
    at /home/chiller/Desktop/git/ddnet/src/engine/client/text.cpp:947
#1  0x00005555556d4995 in CTextRender::TextWidth(void*, float, char const*, int, float, float*, float*)
    (this=0x7ffff7fa9010, pFontSetV=0x0, Size=12, pText=0x7fffffffae73 "999", StrLength=4, LineWidth=112, pAlignedHeight=0x0, pMaxCharacterHeightInLine=0x0) at /home/chiller/Desktop/git/ddnet/src/engine/client/text.cpp:803
#2  0x0000555555745d0f in CMenus::DoEditBox(void*, CUIRect const*, char*, unsigned int, float, float*, bool, int, char const*) (this=
    0x555555ae5120 <gs_Menus>, pID=0x555555af7f48 <g_Config+2984>, pRect=0x7fffffffae60, pStr=0x7fffffffae73 "999", StrSize=5, FontSize=12, Offset=0x55555595ad0c <CMenus::RenderServerbrowserFilters(CUIRect)::Offset>, Hidden=false, Corners=15, pEmptyText=0x5555558ad8d0 "") at /home/chiller/Desktop/git/ddnet/src/game/client/components/menus.cpp:508
#3  0x00005555557586c5 in CMenus::RenderServerbrowserFilters(CUIRect) (this=0x555555ae5120 <gs_Menus>, View=...)
    at /home/chiller/Desktop/git/ddnet/src/game/client/components/menus_browser.cpp:765
#4  0x000055555575c530 in CMenus::RenderServerbrowser(CUIRect) (this=0x555555ae5120 <gs_Menus>, MainView=...)
    at /home/chiller/Desktop/git/ddnet/src/game/client/components/menus_browser.cpp:1409
#5  0x0000555555749688 in CMenus::Render() (this=0x555555ae5120 <gs_Menus>)
    at /home/chiller/Desktop/git/ddnet/src/game/client/components/menus.cpp:1261
#6  0x000055555574fa92 in CMenus::OnRender() (this=0x555555ae5120 <gs_Menus>)
    at /home/chiller/Desktop/git/ddnet/src/game/client/components/menus.cpp:2393
#7  0x00005555557b3a34 in CGameClient::OnRender() (this=0x555555981b80 <g_GameClient>)
    at /home/chiller/Desktop/git/ddnet/src/game/client/gameclient.cpp:676
#8  0x00005555556835fd in CClient::Render() (this=0x7ffff5da4010) at /home/chiller/Desktop/git/ddnet/src/engine/client/client.cpp:1134
#9  0x000055555568bfcf in CClient::Run() (this=0x7ffff5da4010) at /home/chiller/Desktop/git/ddnet/src/engine/client/client.cpp:3290
#10 0x0000555555691c7f in main(int, char const**) (argc=1, argv=0x7fffffffe428)
    at /home/chiller/Desktop/git/ddnet/src/engine/client/client.cpp:4356
```

![image](https://user-images.githubusercontent.com/20344300/99434781-7d65bd80-290f-11eb-86ea-6c4f70d510a4.png)

**Comment by def-** [comment]
Which version? Should be fixed in 15.2.1/15.2.2

**Comment by ChillerDragon** [comment]
Oh duplicate of https://github.com/ddnet/ddnet/pull/3311
